bundles/org.eclipse.equinox.p2.repository/src/org/eclipse/equinox/internal/provisional/p2/repository/DefaultPGPPublicKeyService.java

```diff
@@ -742,9 +742,12 @@ private static abstract class GPGPubringSupplier implements Supplier<PGPPublicKe
 
 		public GPGPubringSupplier(Path pubring) {
 			this.pubring = pubring;
-			
+			try {
 				keyRingCollection = new PGPPublicKeyRingCollection(Collections.emptyList());
-			
+			} catch (Exception e) {
+				// Cannot happen for an empty collection.
+				throw new RuntimeException("Unexpected exception while PGPPublicKeyRingCollection",e);
```

Comment:
I would still check if e is a runtime exception first.

Reply:
While upgrading bouncycastle to 1.78.1 in 4.23 maintenance: 
This is exception I am getting,

[ERROR] 	keyRingCollection = new PGPPublicKeyRingCollection(Collections.emptyList());
[ERROR] 	                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] Unhandled exception type IOException
[ERROR] /lts/git-eclipse/R4_23_maintenance/rt.equinox.p2/bundles/org.eclipse.equinox.p2.repository/src/org/eclipse/equinox/internal/provisional/p2/repository/DefaultPGPPublicKeyService.java:[746] 
[ERROR] 	keyRingCollection = new PGPPublicKeyRingCollection(Collections.emptyList());
[ERROR] 	                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] Unhandled exception type PGPException